controllers/dogu_controller_int_test.go

```diff
@@ -82,10 +76,59 @@ var _ = Describe("Dogu Controller", func() {
 				if err != nil {
 					return false
 				}
-				return true
-			}, timeout, interval).Should(BeTrue())
+				return verifyOwner(createdDogu.Name, service.ObjectMeta)
+			}, interval, timeout).Should(BeTrue())
 			Expect(doguName).To(Equal(service.Name))
 			Expect(namespace).To(Equal(service.Namespace))
+
+			By("Expect created secret")
+			secret := &corev1.Secret{}
+			secretLookupKey := types.NamespacedName{Name: doguName + "-private", Namespace: namespace}
+
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, secretLookupKey, secret)
+				if err != nil {
+					return false
+				}
+				return verifyOwner(createdDogu.Name, secret.ObjectMeta)
+			}, interval, timeout).Should(BeTrue())
+			Expect(doguName + "-private").To(Equal(secret.Name))
+			Expect(namespace).To(Equal(secret.Namespace))
+
+			By("Expect created pvc")
+			pvc := &corev1.PersistentVolumeClaim{}
+
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, doguLookupKey, pvc)
+				if err != nil {
+					return false
+				}
+				return verifyOwner(createdDogu.Name, pvc.ObjectMeta)
+			}, interval, timeout).Should(BeTrue())
+			Expect(doguName).To(Equal(pvc.Name))
+			Expect(namespace).To(Equal(pvc.Namespace))
+
+			By("Delete Dogu")
+			Expect(k8sClient.Delete(ctx, ldapCr)).Should(Succeed())
+
+			dogu := &k8sv1.Dogu{}
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, doguLookupKey, dogu)
+				return apierrors.IsNotFound(err)
+			}, interval, timeout).Should(BeTrue())
 		})
 	})
 })
+
+// verifyOwner checks if the objectmetadata has a specific owner. This method should be used to verify that a dogu is
+// the owner of every related resource. This replaces an integration test for the deletion of dogu related resources.
+// In a real cluster resources without an owner will be garbage collected. In this environment the resources still exist
+// after dogu deletion
+func verifyOwner(name string, obj v1.ObjectMeta) bool {
```

Comment:
This is of course a nice solution to test our implementation. However, in the future, we might need to install Dogus several times in different tests. The objects left after the deletion could possibly make our lives harder.